src/test/java/com/pusher/client/crypto/nacl/SecretBoxOpenerTest.java

```diff
@@ -23,9 +26,56 @@ public void setUp() {
 
     @Test
     public void open() {
-        byte[] clearText = subject.open(cipher, nonce);
+        String clearText = subject.open(cipher, nonce);
+
+        assertThat(clearText).isEqualTo("{\"message\":\"hello world\"}");
+    }
+
+    @Test
+    public void openEmptyMessage() {
+        subject = new SecretBoxOpener(
+                Base64.decode("dwXDg1sGnypM44uPh5Rts/JIP2Y7XkHR5lB/o3rBlVs=")
+        );
+
+        ArrayList<String> nonces = Lists.newArrayList(
+                "p8v9RQR5r6o3G7e2KRgteRi5P90ajKVz",
+                "ZwJVq1LuvUSL5F2EH7RN8P9BcrkxHJS2",
+                "F7ZhEsQLeo/H60T+eghMYdgulNI7iq78",
+                "jU9ShtYe8fz/lOUWnxw7hlb+W5cA6neA",
+                "yGJN1Q03seO6CS8QHRxDLVvetp0hmyCB",
+                "Xi+1aPjEBOTsPATGiEtKnS1YTosvXQQu",
+                "mGi+Rj860MfSGFGbYBwjJfK/kdajUFfg",
+                "YrIvOUf/n6GELtC9GxkccEJsZWdfDXu6",
+                "nRZmojqKgQaPN4KSiIgBfIIk7tU3TuDV"
+        );
+
+        ArrayList<String> ciphers = Lists.newArrayList(
+                "glAlout7BU/REO+kzS+Ixa6uTnV1v0ZdPdRyV4hn",
+                "lbjgrLGKFQJWjaCoTwDV9NxcxSiQ42KdvmPEkN7y",
+                "qFpG2oVtc2hJR99wOZeIjdxaZQUJaSxnS631G521",
+                "Qi8vqaCBoq0iDXhbOFBBSLO6jLsQlfZ1rJP6W8iZ",
+                "QZMiHsoEBGN6GtQQduAMTAdPrspQeLPieCa4+fsH",
+                "BHB7z1oJ51+y790myI6U8Iqtu6BK5CGAw0B6RiNA",
+                "8oh6WxNppFXRQRxE6YHTmipWR6AOlmZe1oIXjQyY",
+                "tfVzEAF0pw7taOmPFWGqfxMHZAYegFr3NHeHQvR3",
+                "+lMeOemGsySbPaBuD2G07gZmofHc1XCfaX/NV4Y6"
+        );
+
+        for (int i = 0; i < ciphers.size(); i++) {
+            try {
+                String clearTest = subject.open(
+                        Base64.decode(ciphers.get(i)),
+                        Base64.decode(nonces.get(i))
+                );
+
+                assertThat(clearTest).isEqualTo("{\"message\":\"\"}");
+            } catch (Exception e) {
+                System.out.println("Failed " + i);
+                System.out.println(e.getLocalizedMessage());
+                assertThat(e).isNull();
```

Comment:
`e` can't be `null` here (would have failed to dereference in the line above).

Now that this test should pass, I would remove the whole try-catch block. I only added it because I had may cases and didn't want the test to stop if one failed (because I knew several would and wanted to know how many). I was more using the test as a playground than an actual test.

Reply:
Hmm. Good point. Have updated 😄